File: 1/js/src/ProductSum.ts
type NestedNumbers = number | NestedNumbers[];

function getProductSum(list: NestedNumbers[]): number {
	return getProductSumHelper(list, 1);
}

function getProductSumHelper(list: NestedNumbers[], depth: number): number {
	let sum = 0;

	for(const item of list) {
		if(typeof item == "number") {
			sum += item;
		} else {
			sum += getProductSumHelper(item, depth+1);
		}
	}

	return sum*depth;
}

(function main() {
	const list: NestedNumbers[] = [5,2, [7,-1], 3, [6,[-13,8],4]];
	console.log(getProductSum(list));
})();

File: 1/java/ProductSum.java
import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

class ProductSum {
	
	public static int getProductSum(List<Object> array, int depth) {
		int sum = 0;	
		for(int i=0; i<array.size(); i++) {
			if(array.get(i) instanceof Integer) {
				sum = sum + ((int)array.get(i));
			} else {
				List<Object> subArray = (List<Object>)array.get(i);
				sum += getProductSum(subArray, depth+1);
			}
			System.out.println(sum);
		}
		return sum*depth;
	}
	
	public static void main(String args[]) {
		List<Object> array = new ArrayList<Object>();
		
		array = Arrays.asList(5,2, new ArrayList<Object>(Arrays.asList(7,-1)), 3, new ArrayList<Object>(Arrays.asList(6, new ArrayList<Object>(Arrays.asList(-13, 8)), 4)));
		
		System.out.println(array.toString());
		System.out.println(getProductSum(array, 1));
	}
}
